evaluator/evaluator/app/routers/dataset.py

```diff
@@ -16,7 +26,185 @@
+    status_code=200,
+)
+async def delete_metadata(
+    dataset_name: str,
+    background_tasks: BackgroundTasks,
+    dataset_handler: DatasetHandler = Depends(DatasetHandler),
+    token_payload: Dict = Depends(auth),
+):
+    validate_access(token_payload)
+    logger.debug(f"Deleting metadata with name: {dataset_name}")
+
+    delete_result = mongo_client.client.evaluator.datasets.delete_one(
+        {"name": dataset_name}
+    )
+    if delete_result.deleted_count == 1:
+        logger.info(
+            f"Metadata entry with dataset_name={dataset_name} successfully deleted. Triggering delete dataset task after HTTP response..."
+        )
+        background_tasks.add_task(trigger_delete_dataset, dataset_handler, dataset_name)
+        return
+    else:
+        message = (
+            f"Dataset {dataset_name} could not be deleted, because it was not found."
+        )
+        logger.warning(message)
+        raise HTTPException(404, message)
+
+
+async def validate_mapping_and_skill_type(dataset_metadata: DatasetMetadata):
+    """
+    Function that checks if the specified skill type exists and if the given mapping has the format of the skill type.
+    """
+    if dataset_metadata.skill_type not in SUPPORTED_SKILL_TYPES.keys():
+        raise HTTPException(
+            400,
+            f'Skill type {dataset_metadata.skill_type} not supported! Please use one of the following skill types: {", ".join(SUPPORTED_SKILL_TYPES.keys())}',
+        )
+    try:
+        SUPPORTED_SKILL_TYPES[dataset_metadata.skill_type].parse_obj(
+            dataset_metadata.mapping
+        )
+    except ValidationError as error:
+        raise HTTPException(
+            400,
+            f"The mapping does not match the skill type {dataset_metadata.skill_type}: {error.errors}",
+        )
+
+
+def validate_access(token_payload: Dict):
+    """
+    Function to validate access. Currently SQuARE has no user roles, so we check for the statically defined user names.
+    """
+    ALLOWED_USER_NAMES = ["ukp", "LOCAL_SQUARE_USER"]
+    try:
+        logger.info(f"USERNAME {token_payload}")
+        if token_payload["username"] not in ALLOWED_USER_NAMES:
+            logger.info(f'Access denied for user_name={token_payload["username"]}')
+            raise KeyError
+        logger.info(f'Access granted for user_name={token_payload["username"]}')
+    except KeyError:
+        raise HTTPException(403, "Forbidden.")
+
+
+def validate_metric(metric: str):
+    if metric_exists(metric) == False:
+        raise HTTPException(
+            400,
+            f"Metric {metric} could not be found.",
+        )
+
+
+def validate_dataset_name(dataset_name: str):
+    if dataset_name_exists(dataset_name) == False:
+        raise HTTPException(
+            400,
+            f"Dataset {dataset_name} could not be found.",
+        )
+
+
+def trigger_download_dataset(dataset_handler: DatasetHandler, dataset_name: str):
+    dataset_handler.download_dataset(dataset_name)
```

Comment:
Wenn ich das richtig sehe, wird jetzt bei jedem Update vom z.B. Mapping des Datasets auch das dataset neu runtergeladen (auch wenn es schon existiert lokal). Würde das also in `dataset_handler.get_dataset()` abändern.

Außer man sagst, das ist ggf. ein Feature?

Reply:
Joa, dachte das ist ein Feature, weil sich die Datasetdaten ja ggf. ändern können (z.B. wenn mehr Datenpunkte hinzugefügt werden). Deswegen lieber immer Neuladen forcen